engine/lib/friends.php

```diff
@@ -204,6 +204,10 @@ function _elgg_setup_collections_menu() {
  * @access private
  */
 function _elgg_send_friend_notification($event, $type, $object) {
+	if ($object->relationship != 'friend') {
+		return true;
```

Comment:
Doesn't matter. Unlike the hooks service, events service does not pass handler return value to the next handler. So returning `true`here simply means that the handler doesn't want to prevent the relationship from being created.